scripts/ghidra/SyncBromaScript.java

```diff
@@ -538,18 +541,25 @@ private void handleExport() throws Exception {
                     bromaFun = bromaFuns.get(0);
                 }
 
-                // Update return type if Ghidra has an user-defined type and 
+                // Update return type if Ghidra has an user-defined type which isn't TodoReturn and 
                 // Broma has TodoReturn
                 if (
                     bromaFun.returnType.isPresent() && bromaFun.returnType.get().name.value.contains("TodoReturn") &&
-                    fun.getReturn().getSource() == SourceType.USER_DEFINED
+                    fun.getReturn().getSource() == SourceType.USER_DEFINED &&
+                    !fun.getReturnType().getDisplayName().contains("TodoReturn")
                 ) {
                     broma.addPatch(bromaFun.returnType.get().range, fun.getReturnType().getDisplayName());
                     exportedTypeCount += 1;
                 }
 
+                // $Sabe: 
+                // Commented this call to importSignatureFromBroma because it's replacing the 
+                // return type in ghidra with the one in the broma file and this is the
+                // export handler, so it effectively undoes the RE work by the user.
+                // Left this comment in case I'm not understanding properly and the call is needed.
+
                 // Get the function signature from Broma
-                importSignatureFromBroma(child.getAddress(), bromaFun, false);
+                // importSignatureFromBroma(child.getAddress(), bromaFun, false);

```

Comment:
The reason for this call is so that if you have RE'd new functions that didn't exist in Broma yet, their signatures will be pulled from Broma (exporting runs after importing so without this call your new functions would be left in their Ghidra-inferred state until you ran the script again)

Reply:
I'd just add a parameter to `importSignatureFromBroma` to mark whether return type should be overridden or not